app/src/main/java/com/example/drawingApp/DrawingViewModel.kt

```diff
@@ -15,8 +15,16 @@ class DrawingViewModel : ViewModel() {
 
     var activeBitmap: Bitmap? = null
 
-    var isLayerCreationDialogOpen = false
-    var currentLayerCreationDialogText = ""
+    var isLayerCreationDialogOpen = false   //new Layer
+    var currentLayerDialogText = ""
 
+    var isLayerViewDialogOpen = false       //edit layer
+    var isLayerViewEditDialogOpen = false   //view inside edit layer
+    var layerViewEditDialogIndex:Int = LAYER_VIEW_EDIT_INDEX
+    var currentViewLayerDialogText = ""
+
+    companion object{
+        const val LAYER_VIEW_EDIT_INDEX = -1
```

Comment:
I like using the companion object, but I think it is interesting you default it to -1 here, but don't have similar constants for layerCounter from the dialog util

I was able to import my companion object in my viewModel into anywhere else I needed the constants, not sure if that would work for you.

More of an idea than a request for a change